### PR TITLE
Attempts to port typescript into our code and updates dependancies

### PR DESCRIPTION
**Pull request by Gboster-0**
I doubt this PR will actually succeed all the checks, but may aswell try

Also this PR needs supervised TMing and extensive admin testing, it may break the fabric of reality

## About The Pull Request

Ports the following PR's:

https://github.com/tgstation/tgstation/pull/57081 (full port)
https://github.com/tgstation/tgstation/pull/57326 (full port)
https://github.com/tgstation/tgstation/pull/57889 (tiny port of typescript collections.ts and math.ts)
https://github.com/tgstation/tgstation/pull/60859 (partial port)
https://github.com/tgstation/tgstation/pull/60961 (partial port)

## Why It's Good For The Game

- Visual Studio can now fix some errors by itself (look at PR #57081)
- And the main thing: we can effortlessly port typescript files from TG (this is the big one, holy shit it was a pain)

## Changelog
:cl:
code: typescript is now compatible with our codebase
/:cl:

### CODE REVIEW

File: tgui/packages/tgui-panel/styles/components/Chat.scss [hunk 1]
```diff
@@ -82,7 +83,7 @@ $color-bg-section: base.$color-bg-section !default;
 
 .ChatMessage--highlighted {
   position: relative;
-  border-left: (1em / 6) solid rgba(255, 221, 68);
+  border-bottom: math.div(1em, 6) solid colors.$red;
```
Comment (LanceSmites328) [inline tgui/packages/tgui-panel/styles/components/Chat.scss:86]: This is why Highlighted messages have a red line under them. Previously this added a Yellow border to the left, now it adds a Red Border to the bottom

**Review by LanceSmites328 — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui-panel/styles/components/Chat.scss (hunk 1 above)
Comment (Gboster-0) [reply tgui/packages/tgui-panel/styles/components/Chat.scss:86]: I have no idea why would you risk your sanity and review the 71 file PR, but neat. Gonna change that

Comment (LanceSmites328) [reply tgui/packages/tgui-panel/styles/components/Chat.scss]: I can not overstate how much I'm supposed to. Also this is kind of HTML? Sort of. I'm somewhat familiar with it

**Comment by LanceSmites328** [comment]
Appears to break paper again.

**Comment by Gboster-0** [comment]
Yep, it does. Honesly i might just close da PR since im not good with TGUI debugging

**Comment by Gboster-0** [comment]
And with this comment, the reception of the typescript has concluded with a failure

If anyone wants to fix/continue working on this feel free to go ahead

**Pull request closed without merging**